### PR TITLE
Boxoffice

### PR DESCRIPTION
**Pull request by shreyas-satish**
Adds support to import orders from https://github.com/hasgeek/boxoffice

### CODE REVIEW

File: funnel/views/event.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def admin(profile, space):
     csrf_form = forms.Form()
     if csrf_form.validate_on_submit():
         for ticket_client in space.ticket_clients:
-            if ticket_client and ticket_client.name == u'explara':
+            if ticket_client and ticket_client.name.lower() in [u'explara', u'boxoffice']:
```
Comment (jace) [inline funnel/views/event.py]: Don't do `lower()` here. It should be validated when saving to the db. Add a migration if you have bad data.

File: funnel/extapi/boxoffice.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    """
+    def __init__(self, access_token):
+        self.access_token = access_token
+        self.base_url = app.config['BOXOFFICE_SERVER']
+
+    def url_for(self, endpoint):
+        return self.base_url.format(endpoint)
+
+    def get_orders(self, ic):
+        resp = requests.get(self.base_url + '/ic/{ic}/orders?access_token={token}'.format(ic=ic, token=self.access_token))
```
Comment (jace) [inline funnel/extapi/boxoffice.py]: What is this access token?

File: funnel/extapi/boxoffice.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        return resp.json().get('orders')
+
+    def get_tickets(self, ic):
+        tickets = []
+        for order in self.get_orders(ic):
+            for line_item in order.get('line_items'):
+                if line_item.get('assignee', {}).get('email'):
+                    tickets.append({
+                        'fullname': line_item.get('assignee', {}).get('fullname', ''),
```
Comment (jace) [inline funnel/extapi/boxoffice.py]: Do you want to sync tickets that don't have an assignee?

File: funnel/jobs/jobs.py [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
     with app.test_request_context():
         ticket_client = TicketClient.query.get(ticket_client_id)
-        if ticket_client and ticket_client.name == u'explara':
-            ticket_list = ExplaraAPI(access_token=ticket_client.client_access_token).get_tickets(ticket_client.client_eventid)
-            # cancelled tickets are excluded from the list returned by get_tickets
-            cancel_list = SyncTicket.exclude(ticket_client, [ticket.get('ticket_no') for ticket in ticket_list]).all()
-            ticket_client.import_from_list(ticket_list, cancel_list=cancel_list)
-        db.session.commit()
+        if ticket_client:
+            if ticket_client.name.lower() == u'explara':
```
Comment (jace) [inline funnel/jobs/jobs.py]: No `.lower()`

File: funnel/jobs/jobs.py [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
+                ticket_list = ExplaraAPI(access_token=ticket_client.client_access_token).get_tickets(ticket_client.client_eventid)
+                # cancelled tickets are excluded from the list returned by get_tickets
+                cancel_list = SyncTicket.exclude(ticket_client, [ticket.get('ticket_no') for ticket in ticket_list]).all()
+                ticket_client.import_from_list(ticket_list, cancel_list=cancel_list)
+            elif ticket_client.name.lower() == u'boxoffice':
```
Comment (jace) [inline funnel/jobs/jobs.py]: No `.lower()`

File: funnel/extapi/boxoffice.py (hunk 2 above)
Comment (shreyas-satish) [inline funnel/extapi/boxoffice.py]: This is the access token obtained from `boxoffice`.

Comment (jace) [inline funnel/extapi/boxoffice.py]: Obtained how? Lastuser should broker all access tokens.

**Comment by vidya-ram** [comment]
Issued another PR #225

**Pull request closed without merging**